src/status_im/contexts/communities/overview/view.cljs

```diff
@@ -279,25 +244,24 @@
       [:<>
        [rn/view {:style style/community-content-container}
         (when-not collapsed?
-          [status-tag pending? joined])
+          [status-tag id joined])
         [community-header name (when collapsed? (get-in images [:thumbnail :uri]))
          (when-not collapsed? description)]
         (when (and (seq tags) (not collapsed?))
           [quo/community-tags
            {:tags            tags
             :last-item-style style/last-community-tag
             :container-style style/community-tag-container}])
-        [join-community community pending?]]
-       (when (or (and (seq token-permissions) joined)
-                 (empty? token-permissions))
+        [join-community community]]
+       (when (or joined (not membership-permissions?))
```

Comment:
FYI: This requirement is captured from:

1. https://www.figma.com/file/h9wo4GipgZURbqqr1vShFN/Communities-for-Mobile?type=design&node-id=15331-284739&mode=design&t=GkPXh1C9jw7RlAe0-0
2. https://www.figma.com/file/h9wo4GipgZURbqqr1vShFN/Communities-for-Mobile?type=design&node-id=11308-429045&mode=design&t=GkPXh1C9jw7RlAe0-0